index/scorch/snapshot_index.go

```diff
@@ -430,13 +432,17 @@ func (i *IndexSnapshot) Document(id string) (rv index.Document, err error) {
 	segmentIndex, localDocNum := i.segmentIndexAndLocalDocNumFromGlobal(docNum)
 
 	rvd := document.NewDocument(id)
-	prevBytesRead := i.segment[segmentIndex].segment.BytesRead()
+	var totalStoredBytes int
 
 	err = i.segment[segmentIndex].VisitDocument(localDocNum, func(name string, typ byte, val []byte, pos []uint64) bool {
 		if name == "_id" {
 			return true
 		}
 
+		// track uncompressed stored fields bytes as part of IO stats.
+		// However, ideally we'd need to track the compressed on-disk value
+		// Keeping that TODO for now until we have a cleaner way.
+		totalStoredBytes += len(val)
```

Comment:
rvd.StoredFieldsBytes += len(val)

Separate question - shouldn't you be doing this only when the field is actually "stored"?

Reply:
This callback is called only for the stored fields. The VisitDocument API ends up calling the VisitStoredFields in the zap layer which loads the bytes pertaining to the stored fields, uncompresses them and iterates over each of these fields and applies this callback.